url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/BitboardPosition.java

```diff
@@ -0,0 +1,2907 @@
+         }
+         else if(s.getRank() == Rank._4)
+         {
+            enPassantSquareDestination[s.index] = s.getPreviousInFile();
+            enPassantSquareSource[s.index] = null;
+         }
+         else if(s.getRank() == Rank._5)
+         {
+            enPassantSquareDestination[s.index] = s.getNextInFile();
+            enPassantSquareSource[s.index] = null;
+         }
+         else
+         {
+            enPassantSquareSource[s.index] = null;
+            enPassantSquareDestination[s.index] = null;
+         }    
+      }
+   } // Fin de iniciarlizar()
+   
+   /**
+    * 
+    *
+    */
+   protected void updateOcupied()
+   {
+      //long ti = System.currentTimeMillis();
+      
+      //Se calcula el bitboard de ocupación de piezas blancas
+      bbWhiteOccupation = bbPiecesOccupation[Piece.WHITE_PAWN.index] |
+                     bbPiecesOccupation[Piece.WHITE_ROOK.index] |
+                     bbPiecesOccupation[Piece.WHITE_KNIGHT.index] |
+                     bbPiecesOccupation[Piece.WHITE_BISHOP.index] |
+                     bbPiecesOccupation[Piece.WHITE_QUEEN.index] |
+                     bbPiecesOccupation[Piece.WHITE_KING.index];
+         
+      //Se calcula el bitboard de ocupación de piezas negras
+      bbBlackOccupation =  bbPiecesOccupation[Piece.BLACK_PAWN.index] |
+                     bbPiecesOccupation[Piece.BLACK_ROOK.index] |
+                     bbPiecesOccupation[Piece.BLACK_KNIGHT.index] |
+                     bbPiecesOccupation[Piece.BLACK_BISHOP.index] |
+                     bbPiecesOccupation[Piece.BLACK_QUEEN.index] |
+                     bbPiecesOccupation[Piece.BLACK_KING.index];
+      
+      bbColourOccupation[Colour.WHITE.index] = bbWhiteOccupation;
+      bbColourOccupation[Colour.BLACK.index] = bbBlackOccupation;
+      
+      //Se calcula el bitboard de las casillas ocupadas
+      bbOccupation = bbWhiteOccupation | bbBlackOccupation;
+      
+      //long to = System.currentTimeMillis();
+      //System.out.println("Tiempo de actualizacion de ocupadas: "+(to-ti)+ " milisegundos");
+      
+   } //Fin de actualizarOcupadas()
+   
+
+   public void updateAttackedFromFull()
+   {  
+      for(int i = 0; i < 64; i++)
+      {
+         bbAttackedFromSquare[i] = 0;
+         bbAttackerToSquare[i] = 0;
+      }
+      
+      
+      //Se generan las casillas atacadas por el rey blanco
+      uaBbOccupiedTemp = bbPiecesOccupation[Piece.WHITE_KING.index];
+      uaNumPieces = Long.bitCount(uaBbOccupiedTemp);
+      for(int i = 0; i < uaNumPieces; i++)
+      {
+         uaSquareIndex = Long.numberOfTrailingZeros(uaBbOccupiedTemp);
+         uaBbAttackerSquare = bbSquare[uaSquareIndex];
+         
+         uaBbAttacked = bbKingAttack[uaSquareIndex];
+         bbAttackedFromSquare[uaSquareIndex] |= uaBbAttacked;
+         
+         updateAttackingTo(uaBbAttackerSquare, uaBbAttacked);
+         
+         uaBbOccupiedTemp ^= uaBbAttackerSquare;
+      }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 1086:1097, 1102:1113, 1118:1129, 1134:1145, 1269:1281, 1286:1298.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.